### DISCUSSION TITLE
includeRoot not working on filterDeep

### DISCUSSION THREAD

**Discussion by GaborTorma** [YuriGor/deepdash#95] [Q&A]
`const tree = {
   id: 1,   
   rootKey: 'rootVal'   
};

let cleaned = _.filterDeep(tree, (k, v) => k === 'rootKey' || v === 'rootVal', {includeRoot: true});

console.log(cleaned);
// get empty object
// expected result: { rootKey: 'rootVal' }`

**Comment by YuriGor** [comment]
Hi @GaborTorma!
Predicate signature is similar to native js array `filter` or Lodash's version:
value goes first, key - second
`let cleaned = _.filterDeep(tree, (v, k) => k === 'rootKey' || v === 'rootVal');`

Also note `includeRoot` makes sense only if you use leavesOnly:false
but in that case you should carefully decide about parent nodes, because if you return false for some parent - children will not be visited.
